v1/retryable_client.go

```diff
@@ -0,0 +1,231 @@
+package authzed
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	v1 "github.com/authzed/authzed-go/proto/authzed/api/v1"
+	"github.com/cenkalti/backoff/v4"
+	"github.com/samber/lo"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"strings"
+	"time"
+)
+
+//----------------------------------------
+
+type ConflictStrategy int
+
+const (
+	Fail ConflictStrategy = iota
+	Skip
+	Touch
+
+	defaultBackoff    = 50 * time.Millisecond
+	defaultMaxRetries = 10
+)
+
+// Fallback for datastore implementations on SpiceDB < 1.29.0 not returning proper gRPC codes
+// Remove once https://github.com/authzed/spicedb/pull/1688 lands
+var (
+	txConflictCodes = []string{
+		"SQLSTATE 23505",     // CockroachDB
+		"Error 1062 (23000)", // MySQL
+	}
+	retryableErrorCodes = []string{
+		"retryable error",                          // CockroachDB, PostgreSQL
+		"try restarting transaction", "Error 1205", // MySQL
+	}
+)
+
+type RetryableClient struct {
```

Comment:
Add doc comments on all exported functions and types

Reply:
I believe I have done this now. If I missed one, let me know.

Reply:
missed one:

> // NewRetryableClient initializes a brand new client for interacting
// with Authzed.

Reply:
Done ✅